packages/ui-kit/src/components/TimeRangePicker/TimeRangePicker.tsx

```diff
@@ -0,0 +1,454 @@
+      caption_start: classNames(styles.caption_start, componentStyles.caption_start),
+      head_cell: classNames(styles.head_cell, componentStyles.head_cell),
+      cell: classNames(styles.cell, componentStyles.cell),
+      button: classNames(styles.button, componentStyles.button),
+      day_today: classNames(styles.day_today, componentStyles.day_today),
+      day_selected: classNames(styles.day_selected, componentStyles.day_selected),
+      day_range_start: classNames(styles.day_range_start, componentStyles.day_range_start),
+      day_range_end: classNames(styles.day_range_end, componentStyles.day_range_end),
+      day_range_middle: classNames(styles.day_range_middle, componentStyles.day_range_middle),
+      multiple_months: classNames(styles.multiple_months, componentStyles.multiple_months)
+    }
+
+    return (
+      <div ref={setRef}>
+        <Select
+          disabled={disabled || (disableOptions && disableCustomRelative && disableCustomRange && disableDateUntilNow)}
+          startAdornment={() => <CalendarIcon size={16} />}
+          value={selectedOptionLabel}
+          placeholder="Date Range"
+          onListboxOpenChange={setSelectOpen}
+          listboxOpen={selectOpen}
+          size="small"
+          slotProps={{
+            popper: { className: componentStyles.timeRangePickerPopper }
+          }}
+          onChange={(event, value) => {
+            if (value === FROM_DATE_UNTIL_NOW || value === 'custom-fixed-date-range') {
+              event?.stopPropagation()
+              setSelectOpen(false)
+              setAnchorEl(componentContainer)
+            }
+
+            if (value === FROM_DATE_UNTIL_NOW) {
+              const now = new Date()
+              setSelectedRange({ from: startOfDay(now), to: now })
+              if (!datepickerRange?.to) {
+                setDatepickerRange({ from: undefined, to: now })
+              }
+              setDatepickerMode(FROM_DATE_UNTIL_NOW)
+              return
+            }
+
+            if (value === 'custom-fixed-date-range') {
+              setDatepickerMode(CUSTOM_DATE_RANGE)
+              return
+            }
+
+            if (value === 'last-n') {
+              event?.stopPropagation()
+              setSelectedOption(getLastNOption())
+              return
+            }
+
+            setSelectedOption(options?.find((option) => option.uid === value))
+          }}
+        >
+          {!disableOptions &&
+            options?.map((option) => (
+              <Option key={option.uid} value={option.uid}>
+                {option.label}
+              </Option>
+            ))}
+          {!disableCustomRelative && (
+            <Option value="last-n" className={componentStyles.lastN}>
+              In the last
+              {/* eslint-disable-next-line jsx-a11y/no-static-element-interactions, jsx-a11y/click-events-have-key-events */}
+              <div onClick={(event) => event.stopPropagation()}>
```

Comment:
Is this `stopPropagation` necessary? I tried removing it and didn't seem to make any effect, we should listen to those eslint warnings instead of snoozing them since this one for example seems to affect a11y

Reply:
This prevents firing select logic instantly on the parent `Option` component when user change values in the nested inputs.